### PR TITLE
Update goose dependency

### PR DESCRIPTION
**Pull request by mhemmings**
Update goose for swift fixes. Makes a couple of error tests a bit less strict.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore/2249/

File: internal/blobstore/blobstore_test.go [hunk 1]
```diff
@@ -279,7 +279,7 @@ func (s *blobStoreSuite) TestPutPartAgain(c *gc.C) {
 	// Perform a Put with mismatching content. This should leave the part in progress
 	// but not completed.
 	err := s.store.PutPart(id, 0, strings.NewReader("something different"), int64(len(content)), 0, hashOf(content))
-	c.Assert(err, gc.ErrorMatches, `cannot upload part ".+": hash mismatch`)
```
Comment (frankban) [inline internal/blobstore/blobstore_test.go:282]: I guess the new version of goose does not allow us to be a bit more specific when checking the error?

**Review by frankban — APPROVED** [review]
Looks good, thanks!

**Comment by rogpeppe** [comment]
LGTM

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**